tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -456,6 +485,34 @@ def test_codecov_line_coverage_exception(self, mock_integration, mock_get_codeco
             )
         ]
 
+    @mock.patch("sentry.integrations.utils.codecov.get_codecov_token", return_value="token")
+    @mock.patch("sentry.integrations.utils.codecov.requests.get", side_effect=mocked_requests_get)
+    def test_get_codecov_data_with_sha(self, mock_token, mock_get):
+        line_coverage, codecov_url = get_codecov_data(
+            repo="getsentry/sentry",
+            service="github",
+            ref="93be0c7da1f291e85e8de167276f21a854fe4590",
+            ref_type="sha",
+            branch="master",
+            path="src/sentry/integrations/github/client.py",
+        )
+        assert line_coverage == expected_line_coverage
+        assert codecov_url == expected_codecov_url
```

Comment:
Create a helper function instead of using `expected_codecov_url`.

Something like this:
```
def codecov_app_url_helper(service='gh', repo_full_name, ref, ref_type, path):
    // XXX: more here
    return f"https://app.codecov.io/{service}/{repo_full_name}/{something_here}/{path}"
```

I would expect this to be (user helper function here):
```
assert codecov_url == "https://app.codecov.io/gh/getsentry/sentry/commit/93be0c7da1f291e85e8de167276f21a854fe4590/blob/src/sentry/integrations/github/client.py"
```